rust/src/live.rs

```diff
@@ -301,50 +302,33 @@ fn update_etc(
 // we actually need to escape our mount namespace and affect
 // the "main" mount namespace so that other processes will
 // see the overlayfs.
+#[context("Creating overlayfs")]
 fn unlock_transient(sysroot: &ostree::Sysroot) -> Result<()> {
     // Temporarily drop the lock
     sysroot.unlock();
-    let status = Command::new("systemd-run")
-        .args(&[
-            "-u",
-            "rpm-ostree-unlock",
-            "--wait",
-            "--",
-            "ostree",
-            "admin",
-            "unlock",
-            "--transient",
-        ])
-        .status();
-    sysroot.lock()?;
-    let status = status?;
-    if !status.success() {
-        bail!("Failed to unlock --transient");
-    }
+    isolation::run_systemd_worker_sync(&isolation::UnitConfig {
+        name: Some("rpm-ostree-unlock"),
+        properties: &[],
+        exec_args: &["ostree", "admin", "unlock", "--transient"],
+    })?;
     Ok(())
 }
 
-/// Run `systemd-tmpfiles` via `systemd-run` so we escape our mount namespace.
-/// This allows our `ProtectHome=` in the unit file to work.
+/// Run `systemd-tmpfiles` as a separate systemd unit to escape
+/// our mount namespace.
+/// This allows our `ProtectHome=` in the unit file to work
+/// for example.  Longer term I'd like to protect even more of `/var`.
+#[context("Running tmpfiles for /run and /var")]
 fn rerun_tmpfiles() -> Result<()> {
-    for prefix in &["/run", "/var"] {
-        let status = Command::new("systemd-run")
-            .args(&[
-                "-u",
-                "rpm-ostree-tmpfiles",
-                "--wait",
-                "--",
-                "systemd-tmpfiles",
-                "--create",
-                "--prefix",
-                prefix,
-            ])
-            .status()?;
-        if !status.success() {
-            bail!("Failed to invoke systemd-tmpfiles");
-        }
-    }
-    Ok(())
+    isolation::run_systemd_worker_sync(&isolation::UnitConfig {
+        name: Some("rpm-ostree-tmpfiles"),
+        properties: &[],
+        exec_args: &[
+            "/bin/sh",
+            "-c",
+            "systemd-tmpfiles --create --prefix /run && systemd-tmpfiles --create --prefix /var",
+        ],
```

Comment:
Have you tried

```suggestion
        exec_args: &[
            "systemd-tmpfiles",
            "--create",
            "--prefix=/run",
            "--prefix=/var",
        ],
```

?